Sources/Variable.swift

```diff
@@ -100,6 +100,20 @@ public struct Variable : Equatable, Resolvable {
         } else if bit == "count" {
           current = array.count
         }
+      } else if let string = current as? String {
```

Comment:
Why repeat this whole `else if` branch, if you can just use the if branch above?

Reply:
I tried to reuse it but you can't interchangeably use `[Any]` and `String`, using available collections type erasures causes other issues like accessing via subscript

Reply:
We could use `string.character` but it will give deprecation warning

Reply:
How about extracting the branch into something like `func resolve<T: RandomAccessCollection>(items: T, bit: String) {`?

Reply:
yes, this will probably work

Reply:
Quick test in playgrounds:
```swift
func resolve<T: Collection>(items: T, bit: String) -> Any? {
	if let index = Int(bit) {
		if index >= 0 && index < items.count {
			return items[items.index(items.startIndex, offsetBy: index)]
		} else {
			return nil
		}
	} else if bit == "first" {
		return items.first
	} else if bit == "last" {
		return items[items.index(items.endIndex, offsetBy: -1)]
	} else if bit == "count" {
		return items.count
	} else {
		return nil
	}
}
```